admin-part/src/main/kotlin/Plugin.kt

```diff
@@ -550,28 +571,29 @@ class Plugin(
         }?.close()
     }
 
-
-    internal fun BundleCounters.assocTestsJob(
+    internal suspend fun BundleCounters.assocTestsJob(
         scope: Scope? = null,
-    ) = GlobalScope.launch {
+    ) = PluginPoolScope.launch {
         trackTime("assocTestsJob") {
             logger.debug { "Calculating all associated tests..." }
             val assocTestsMap = byTest.associatedTests(onlyPackages = false)
-            val associatedTests: List<AssociatedTests> = measureTimedValue {
+            val associatedTests = trackTime("assocTestsJob getAssociatedTests") {
                 assocTestsMap.getAssociatedTests()
-            }.apply { logger.trace { "Calculated in ${duration.inSeconds}" } }.value
+            }
             val treeCoverage = state.coverContext().packageTree.packages.treeCoverage(all, assocTestsMap)
             logger.debug { "Sending all associated tests" }
             scope?.let {
-                sendScopeTree(it.id, associatedTests, treeCoverage)
-            } ?: sendBuildTree(treeCoverage, associatedTests)
+                trackTime("assocTestsJob sendScopeTree") {
```

Comment:
You already added log in sendScopeTree and sendBuildTree methods

Reply:
Example of log:
> 2021-05-25 12:38:40.965 [pool-1-thread-5] DEBUG plugins//test2code/util/PerfLoggerKt - [sendScopeTree associatedTests count: 6366] took: 238ms
2021-05-25 12:38:42.636 [pool-1-thread-3] DEBUG plugins//test2code/util/PerfLoggerKt - [sendScopeTree packages size: 700] took: 1.63ms
2021-05-25 12:38:42.636 [pool-1-thread-3] WARN  plugins//test2code/util/PerfLoggerKt - [assocTestsJob sendScopeTree] took: 1.91s

as you can see the logs in method sendScopeTree are not equal to full method